subprojects/core/src/main/java/org/gradle/api/internal/DependencyClassPathProvider.java

```diff
@@ -57,7 +68,7 @@ private ClassPath gradleApi() {
         for (Module pluginModule : pluginModuleRegistry.getApiModules()) {
             classpath = classpath.plus(pluginModule.getClasspath());
         }
-        return classpath;
+        gradleApi = classpath;
```

Comment:
:x: Move the assignment to the `if` block within `gradleApi()` so each method has a single responsibility.

Reply:
Done